xdem/coreg/biascorr.py

```diff
@@ -839,48 +1058,87 @@ def __init__(
         self._meta["poly_order"] = poly_order
         self._needs_vars = False
 
-    def _fit_func(  # type: ignore
+    def _fit_rst_rst(  # type: ignore
+        self,
+        ref_elev: NDArrayf,
+        tba_elev: NDArrayf,
+        inlier_mask: NDArrayb,
+        transform: rio.transform.Affine,
+        crs: rio.crs.CRS,
+        z_name: str,
+        bias_vars: dict[str, NDArrayf] | None = None,
+        weights: None | NDArrayf = None,
+        verbose: bool = False,
+        **kwargs,
+    ) -> None:
+
+        # The number of parameters in the first guess defines the polynomial order when calling np.polyval2d
+        p0 = np.ones(shape=((self._meta["poly_order"] + 1) * (self._meta["poly_order"] + 1)))
```

Comment:
why not use square?

Reply:
I don't know :see_no_evil:, haha. I'll try with a square I guess it should still work the same.